frontend/src/component/changeRequest/ChangeRequest/Changes/Change/ChangeOverwriteWarning.tsx

```diff
@@ -137,10 +139,19 @@ const DetailsTable: React.FC<{
     );
 };
 
-const OverwriteWarning: React.FC<{
+export const OverwriteWarning: React.FC<{
```

Comment:
Understood. But this file has three primary exported components:
- EnvVariantChangesToOverwrite
- SegmentChangesToOverwrite
- StrategyChangesToOverwrite

Additionally, the OverwriteWarning component is exported to make it testable (because it relies on API hooks which make unit testing difficult otherwise).

So to me, ChangeOverwriteWarning sounds like a reasonable name for the file. However, if we want to strictly adhere to "only one exported component per file", then I could break it up into four separate files if we think that's better? We could also refactor the three main entrypoints to be one main entrypoint instead (toyed a little bit around with that in https://github.com/Unleash/unleash/pull/6215).

Do you have any preferences?

Reply:
Oh, and if you'd like to split this into multiple files, would you be alright with doing that in a follow-up PR so that we can get this merged and pause the project? It feels like a refactoring job isn't contributing to the project, so we can do that as and when we can.

Reply:
Yes, I'd like a file not to export more than one component because when you work in that context you need to maintain a mental mapping that the file contains the component. If every exported component has it's own file with it's own name then navigating the components is a lot easier because the heuristic allows you to drop the mental mapping and reduce the cognitive overhead. @thomasheartman

Reply:
Addressed in the recent changes. I moved the three separate components into a single entrypoint (as done in the previously linked PR) and moved all the related files into a separate subdirectory.